File: app/src/main/java/me/ykrank/s1next/viewmodel/EmoticonViewModel.java
package me.ykrank.s1next.viewmodel;

import android.databinding.ObservableField;
import android.view.View;

import com.github.ykrank.androidtools.widget.RxBus;

import me.ykrank.s1next.data.api.model.Emoticon;
import me.ykrank.s1next.view.event.EmoticonClickEvent;

public final class EmoticonViewModel {

    public final ObservableField<Emoticon> emoticon = new ObservableField<>();

    public View.OnClickListener clickEmotion(RxBus rxBus) {
        return v -> {
            // notify ReplyFragment that emoticon had been clicked
            rxBus.post(new EmoticonClickEvent(emoticon.get().getEntity()));
        };
    }
}

File: app/src/main/java/me/ykrank/s1next/viewmodel/EmoticonViewModel.kt
package me.ykrank.s1next.viewmodel

import androidx.databinding.ObservableField
import android.view.View

import com.github.ykrank.androidtools.widget.EventBus

import me.ykrank.s1next.data.api.model.Emoticon
import me.ykrank.s1next.view.event.EmoticonClickEvent

class EmoticonViewModel {

    val emoticon = ObservableField<Emoticon>()

    fun clickEmotion(eventBus: EventBus): View.OnClickListener {
        return View.OnClickListener { view ->
            emoticon.get()?.let {
                // notify ReplyFragment that emoticon had been clicked
                eventBus.postDefault(EmoticonClickEvent(it.entity))
            }
        }
    }
}
